mate_menu/plugins/applications.py

```diff
@@ -1551,13 +1577,13 @@ def buildCategoryList( self ):
         num = 1
 
         for menu in self.menuFiles:
-            for child in menu.directory.get_contents():
-                if child.get_type() == matemenu.TYPE_DIRECTORY:
-                    icon =  str(child.icon)
+            for child in get_contents(menu.directory):
+                if isinstance(child, MateMenu.TreeDirectory):
+                    #icon =  child.get_icon().to_string()
```

Comment:
Hmm... the `icon` variable can be reused in the `append` function call later, to avoid another `get_icon().to_string()` conversion.
Also you can add something like `name = child.get_name()` here and then reuse the `name` variable in the `append` call too.